src/server/req_quejob.c

```diff
@@ -285,26 +285,16 @@ generate_objid(char *idbuf, char *clusterid, int objtype, char resv_char)
 {
 	static int svr_id = -1;
 
-	if (idbuf == NULL || server_name == NULL)
+	if (idbuf == NULL || server_name == NULL || clusterid == NULL)
 		return 1;
 
 	if (get_num_servers() <= 1) { /* single server setup */
-		if (clusterid != NULL) {
-			if (objtype == 0)
-				sprintf(idbuf, "%lld.%s", next_svr_sequence_id, clusterid);
-			else if (objtype == 1)
-				sprintf(idbuf, "%lld[].%s", next_svr_sequence_id, clusterid);
-			else if (objtype == 2)
-				sprintf(idbuf, "%c%lld.%s", resv_char, next_svr_sequence_id, clusterid);
-		} else {
-			if (objtype == 0)
-				sprintf(idbuf, "%lld", next_svr_sequence_id);
-			else if (objtype == 1)
-				sprintf(idbuf, "%lld[]", next_svr_sequence_id);
-			else if (objtype == 2)
-				sprintf(idbuf, "%c%lld", resv_char, next_svr_sequence_id);
-		}
-
+		if (objtype == 0)
+			sprintf(idbuf, "%lld.%s", next_svr_sequence_id, clusterid);
+		else if (objtype == 1)
```

Comment:
Can you use macros for objtype? There are macros defined in pbs_ifl.h for object types.

Reply:
There were no macros used earlier. May be because it used to act on only job/array job. Now that we introduced the reservation object also into this function, it would be better to use a macro. As discussed I am now making use of the macros MGR_OBJ_JOB/MGR_OBJ_JOBARRAY_PARENT/MGR_OBJ_RESV which makes it more readable.